client/cmd.go

```diff
@@ -373,3 +378,40 @@ func SetCmdClientContext(cmd *cobra.Command, clientCtx Context) error {
 
 	return nil
 }
+
```

Comment:
I think for backward compatibility we should update GetServerContextFromCmd to use those and reconstruct a legacy server context?

Reply:
So our cmd context will keep both viper, logger and legacyServerCtx?

Reply:
Basically, `GetServerContextFromCmd` instead of reading the server context, it should read the viper context and the logger context and reconstruct a server context. This way it makes use of the latest behavior, without breaking existing users.

So, all the changes here are good, we just need to update `GetServerContextFromCmd` so it keeps working

Reply:
make sense, Im following this way

Reply:
Resolved by https://github.com/cosmos/cosmos-sdk/pull/20422/commits/b31896d9dbfbfcf67e18a077ddeb04c7c66acb64